### PR TITLE
Connection retry from dkeycache when dkeyserver is unavailable

### PR DESCRIPTION
**Pull request by ElvisXYKun**
Signed-off-by: xuyangkun <qiangx.pan@intel.com>

### CODE REVIEW

File: dkeycache/App/main.cpp [hunk 1]
```diff
@@ -59,7 +59,27 @@ int ocall_socket(int domain, int type, int protocol)
...
+    int32_t retry_count = 360;
+    do
+    {
+        int ret = -1;
+        ret = connect(sockfd, servaddr, addrlen);
+        if (ret >= 0)
+        {
+            return ret;
+        }
+        else if (retry_count > 0)
```
Comment (yang8621) [inline dkeycache/App/main.cpp:71]: 换一行 用if

File: dkeycache/App/main.cpp [hunk 2]
```diff
@@ -59,7 +59,27 @@ int ocall_socket(int domain, int type, int protocol)
 
 int ocall_connect(int sockfd, const struct sockaddr *servaddr, socklen_t addrlen)
 {
-    return connect(sockfd, servaddr, addrlen);
+    int32_t retry_count = 360;
```
Comment (yang8621) [inline dkeycache/App/main.cpp:62]: 60就行了

File: dkeycache/App/main.cpp [hunk 3]
```diff
@@ -59,7 +59,27 @@ int ocall_socket(int domain, int type, int protocol)
...
+            log_i("Failed to Connect dkeyserver, sleep 0.5s and try again...\n");
+            usleep(500000); // 0.5 s
+        }
+        else
+        {
+            log_e("Failed to connect dkeyserver\n");
+            return -1;
+        }
+    } while (retry_count-- > 0);
+    return -1;
```
Comment (yang8621) [inline dkeycache/App/main.cpp:82]: do {
int ret = connect(sockfd, servaddr, addrlen);
if (ret >= 0)
    return ret;

log_i("Failed to Connect dkeyserver, sleep 0.5s and try again...\n");
usleep(500000); // 0.5s
} while (retry_count-- > 0);

log_e("Failed to connect dkeyserver.\n");
return -1;

**Comment by yang8621** [comment]
change title to:
connection retry from dkeycache when dkeyserver is unavailable

Passed all the unit test. -- 这个删了，这个没有unittest。

File: dkeycache/App/main.cpp (hunk 2 above)
Comment (syan10) [reply dkeycache/App/main.cpp:62]: 10次吧

File: dkeycache/App/main.cpp (hunk 1 above)
Comment (ElvisXYKun) [reply dkeycache/App/main.cpp]: 好的，修改完毕

File: dkeycache/App/main.cpp (hunk 2 above)
Comment (ElvisXYKun) [reply dkeycache/App/main.cpp]: 好的，修改完毕

File: dkeycache/App/main.cpp (hunk 3 above)
Comment (ElvisXYKun) [reply dkeycache/App/main.cpp:74]: 好的，修改完毕

**Review by syan10 — APPROVED** [review]

**Pull request merged**